### PR TITLE
Add tests for raw API endpoints

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

**Review by Thoralf-M — APPROVED** [review]

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Test fails

**Review by Thoralf-M — APPROVED** [review]

File: sdk/tests/client/node_api.rs [hunk 1]
```diff
@@ -287,13 +287,17 @@ async fn test_get_milestone_by_id_raw() {
...
     let milestone = client.get_milestone_by_id(&latest_milestone_id).await.unwrap();
-    let milestone_raw = MilestonePayload::unpack_verified(
+    let milestone_raw = Payload::unpack_verified(
         client.get_milestone_by_id_raw(&latest_milestone_id).await.unwrap(),
         &client.get_protocol_parameters().await.unwrap(),
     )
     .unwrap();
 
-    assert_eq!(milestone, milestone_raw);
+    if let Payload::Milestone(milestone_raw) = milestone_raw {
```
Comment (DaughterOfMars) [inline sdk/tests/client/node_api.rs:296]: So one of these returns a `MilestonePayload` and one returns `Payload`? Why is that?

Comment (thibault-martinez) [reply sdk/tests/client/node_api.rs:296]: Comes from a difference between the SDK and iota.go I guess, they don't have a payload enum so the type is on the variant

Comment (thibault-martinez) [reply sdk/tests/client/node_api.rs:296]: I don't think we should modify the bytes returned by the API to remove the first one

Comment (DaughterOfMars) [reply sdk/tests/client/node_api.rs:296]: Alright 🤷‍♀️

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**